src/AccordionWithHeader.js

```diff
@@ -60,6 +60,17 @@ export default class AccordionWithHeader extends Component {
     }
   }
 
+  componentDidUpdate() {
```

Comment:
```suggestion
  componentDidUpdate() {
    // in the case of new <AccordionNode/>'s being added or removed
    const hasNewChildren =
      this.props.children.length !== this.state.children.length;
    if (hasNewChildren) {
      // redefine the number of AccordionNode "panels" to control
      const panels = Children.map(this.props.children, (child) => +child.key);
      if (panels.length > 0) {
        this.setState({ ...this.props, panels });
      } else {
        console.warn("Should not have zero panels. your Accordion is gone!");
      }
    }
  }
  
```
This way it will safely account nodes being removed, and throw a warning if all nodes are removed, per @mklengs suggestion.

Let me know what you think @wbenevides

Reply:
Here's a demo of the changes: http://better-coal.surge.sh/  see the Simple example and buttons